### PR TITLE
Update Firebolt

### PR DESCRIPTION
**Pull request by Frenoss**

### CODE REVIEW

File: src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java [hunk 1]
```diff
@@ -62,7 +63,7 @@ public void use(AbstractPlayer p, AbstractMonster m) {
     public void upgrade() {
         if (!upgraded) {
             upgradeName();
-            upgradeDescription();
+            upgradeMagicNumber(UPGRADE_PLUS_PER_CLARITY);
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: Needs to keep an `initializeDescription()` still; that's what triggers the `!M!` placeholder to update immediately in the description

File: src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java [hunk 2]
```diff
@@ -30,7 +30,8 @@
 
     private static final int COST = 1;
     private static final int DAMAGE = 8;
-    private static final int AMOUNT_PER_CLARITY = 2;
+    private static final int AMOUNT_PER_CLARITY = 1;
+    private static final int UPGRADE_PLUS_PER_CLARITY = 1;
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: I think this needs a more significant update to remove `calculateBonusBaseDamage` and remove the `if(upgraded)` conditional from the burning effect in `use()`

File: src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java (hunk 1 above)
Comment (Frenoss) [reply src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: Hmm, in the few different upgrade views I've tried, I always see the upgraded number. Do you have a repro?

File: src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java (hunk 2 above)
Comment (Frenoss) [reply src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: Yes it does. Oy.

File: src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java (hunk 1 above)
Comment (dbjorge) [reply src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: I don't remember offhand, sorry, I just remember having to fix this before for other cards; I think it was something obscure like "when an upgraded card shows up in a shop" or "when the tongs relic upgrades a card" or something

Comment (Frenoss) [reply src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: I'm sorry to be stuck on this, but I really do wonder if something's going on there or what. I have now tried all of the relics that mention upgrading attacks, and they all look to have worked correctly. I think I could reduce the card pool and try out the merchant problem.

I guess a big part of my reluctance is that the base game's cards don't modify their description unless text changes. That makes it seem like the right way to do it.

Comment (dbjorge) [reply src/main/java/stsjorbsmod/cards/wanderer/Firebolt.java]: If the base game doesn't do it, I'm probably misremembering. Would like to update the rest of the cards to be consistent, though, if it's missing in just one spot it looks like a bug.

**Pull request merged**